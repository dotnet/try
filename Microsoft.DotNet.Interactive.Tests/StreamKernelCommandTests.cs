// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System.Diagnostics;
using Assent;
using Microsoft.DotNet.Interactive.Commands;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;
using Xunit;

namespace Microsoft.DotNet.Interactive.Tests
{
    public class StreamKernelCommandTests
    {
        private static readonly JsonSerializerSettings _jsonSerializerSettings = new JsonSerializerSettings
        {
            ContractResolver = new CamelCasePropertyNamesContractResolver()
        };

        private readonly Configuration _configuration;
        public StreamKernelCommandTests()
        {
            _configuration = new Configuration()
                .UsingExtension("json");
            _configuration = _configuration.SetInteractive(Debugger.IsAttached);
        }

        [Fact]
        public void Serialization_contract_is_not_broken()
        {
            var command = new SubmitCode("display(x); display(x + 1); display(x + 2);");
            var streamCommand =  new StreamKernelCommand
            {
                Id = 123,
                CommandType = command.GetType().Name,
                Command = command
            };

            var serializedCommand = JsonConvert.SerializeObject(streamCommand, _jsonSerializerSettings);
            this.Assent(serializedCommand, _configuration);
        }
    }
}