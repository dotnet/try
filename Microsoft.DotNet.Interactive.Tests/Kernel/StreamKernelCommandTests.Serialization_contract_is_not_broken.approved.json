{"id":123,"commandType":"SubmitCode","command":{"code":"display(x); display(x + 1); display(x + 2);","targetKernelName":null,"submissionType":0}}